OpenRA.Mods.Common/Traits/BotModules/UnitBuilderBotModule.cs

```diff
@@ -86,8 +86,11 @@ void IBotTick.BotTick(IBot bot)
 					queuedBuildRequests.Remove(buildRequest);
 				}
 
-				foreach (var q in Info.UnitQueues)
-					BuildUnit(bot, q, idleUnitCount < Info.IdleBaseUnitsMaximum);
+				if (Info.IdleBaseUnitsMaximum > idleUnitCount)
```

Comment:
won't this cause unit building to stall? dead lock - while base units remain idle?

Reply:
I think it is normal. 
1. `idleUnitCount` are updated by `SquadManagerBotModule`, those whom join squad will be removed from idle count. You can set `IdleBaseUnitsMaximum` always bigger than "squadsize + unit exclude from squad" to avoid this.

2. Even for old code here, when `Info.IdleBaseUnitsMaximum > idleUnitCount` the code will use `ChooseUnitToBuild`, which is a costy function and also has possibility to build nothing in the end.

3. If `IdleBaseUnitsMaximum` cannot control and prevent more units to be built, then I really don't see how it can help on "only produce units as long as there are less than this amount of units idling inside the base", especially for game performance purpose.